src/reference/zilla.yaml/binding-grpc-kafka.md

```diff
@@ -0,0 +1,414 @@
+
+#### options.correlation
+
+> `object`
+
+Kafka request message headers injected when adapting `grpc` request-response streams to `kafka` topic streams.
+
+##### correlation.headers
+
+> `object`
+
+Kafka request message reply to and correlation id header names injected when adapting `grpc` request-response streams to `kafka` topic streams.
+
+##### headers.service
+
+> `string`
+
+Kafka header name for `grpc` service.\
+Defaults to `"zilla:service"`.
+
+##### headers.method
+
+> `string`
+
+Kafka header name for `grpc` method.\
+Defaults to `"zilla:method"`.
+
+##### headers.correlation-id
+
+> `string`
+
+Kafka header name for request-response correlation identifier.\
+Defaults to `"zilla:correlation-id"`.
+
+##### headers.reply-to
+
+> `string`
+
+Kafka header name for reply-to topic.\
+Defaults to `"zilla:reply-to"`.
+
+### routes
+
+> `array` of `object`
+
+Conditional `grpc-kafka`-specific routes for adapting `grpc` request-response streams to `kafka` topic streams.
+
+```yaml
+routes:
+  - guarded:
+      test0:
+        - read:messages
+    when:
+      - service: example.FanoutService
+        metadata:
+          custom-text: custom value
+          custom-binary:
+            base64: Y3VzdG9tIHZhbHVl
+    exit: kafka_cache_client0
+    with:
+      capability: fetch
+      topic: messages
+      filters:
+        key: custom-key
+        headers:
+          custom-text: custom-value
+  - guarded:
+      test0:
+        - echo:messages
+    when:
+      - method: example.EchoService/*
+        metadata:
+          custom-text: custom value
+          custom-binary:
+            base64: Y3VzdG9tIHZhbHVl
+    exit: kafka_cache_client0
+    with:
+      capability: produce
+      topic: requests
+      acks: leader_only
+      key: custom-key
+      overrides:
+        custom-text: custom-value
+      reply-to: responses
+```
+
+### routes[].guarded
+
+> `object` as named map of `string:string` `array`
+
+Roles required by named guard.
+
+```yaml
+routes:
+  - guarded:
+      test0:
+        - read:messages
+```
+
+### routes[].when
+
+> `array` of `object`
+
+List of conditions (any match) to match this route when adapting `grpc` request-response streams to `kafka` topic streams.
+
+```yaml
+routes:
+  - when:
+      - method: example.EchoService/*
+        metadata:
+          custom-text: custom value
+          custom-binary:
+            base64: Y3VzdG9tIHZhbHVl
+```
+
+#### when[].method
+
+> `string`
+
+Pattern matching the fully qualified name of a `grpc` service method, in the format `<service>/<method>` allowing wildcard `*` for method to indicate any method.
+
+#### when[].metadata
+
+>  `object` of name-value headers
+
+Metadata header name value pairs (all match).
+
+Each metadata header value can be `string` or `object` with `base64` property.
+
+##### metadata.base64
+
+> `string`
+
+Base64 encoded value for binary metadata header.
+
+### routes[].exit\*
+
+> `string`
+
+Routed exit binding when conditional route matches.
```

Comment:
```suggestion
Next binding when following this route.
```

The other descriptions of this property use the suggested text. Should we update them all to be consistent, or is this case different?